### ISSUE TITLE
Bad management of pairs of braces

### ISSUE THREAD

**Issue by Heziode** [motdotla/dotenv-expand#49]
Input: `TEST_WITH_BRACKET="{ port: $MONGOLAB_PORT}"`

**Expected**: `{ port: 12345}`

**Actual**: `{ port: 12345`

**Comment by Heziode** [comment]
Possible solution:

Change this line:
https://github.com/motdotla/dotenv-expand/blob/de9e5cb0215495452f475f5be4dea1580b8217cd/lib/main.js#L8

By:
```js
var matches = envValue.match(/(.?\$(?:{(?:[a-zA-Z0-9_]+)?}?|(?:[a-zA-Z0-9_]+)?))/g) || []
```

**Comment by nVitius** [comment]
Coincidentally, the code I wrote in #48 also fixes this issue.
I've added a test for it here: https://github.com/motdotla/dotenv-expand/pull/48/commits/d6f3d97906c7f089cca953e99788fd0ebde5a482

The issue was that the original regex was conditionally looking for the curly brackets. Which means that `${var` and `$var}` would have matched it. My fix expands the regex a bit to explicitly look for strings that have both or neither (as well as `$(command)`.
